pkg/kubelet/kubelet.go

```diff
@@ -1806,10 +1806,16 @@ func (kl *Kubelet) syncTerminatingPod(ctx context.Context, pod *v1.Pod, podStatu
 		}
 		containers = append(containers, fmt.Sprintf("(%s state=%s exitCode=%d finishedAt=%s)", s.Name, s.State, s.ExitCode, s.FinishedAt.UTC().Format(time.RFC3339Nano)))
 	}
-	if klog.V(4).Enabled() {
-		sort.Strings(containers)
-		klog.InfoS("Post-termination container state", "pod", klog.KObj(pod), "podUID", pod.UID, "containers", strings.Join(containers, " "))
-	}
+	klog.V(4).InfoS("Post-termination container state", "pod", klog.KObj(pod), "podUID", pod.UID, "containers", klog.As(
```

Comment:
Another example for `klog.As`.

Reply:
This example definitely feels a bit verbose for a log line, although it seems it's best solution we can get today for the requirements. I can't think of a better solution, maybe something that could be revisited if/when we adopt Go generics

Reply:
The newer revision uses a custom struct and logs a slice of that in text and JSON output.